core/domain/exp_services.py

```diff
@@ -1723,6 +1723,496 @@ def is_voiceover_change_list(change_list):
+                                change.property_name)
+                    else:
+                        changed_properties[state_name] = [change.property_name]
+                elif change.cmd == exp_domain.CMD_ADD_WRITTEN_TRANSLATION:
+                    changed_property = None
+                    if change.content_id == 'content':
+                        changed_property = 'content'
+                    elif change.content_id[:14] == 'ca_placeholder':
+                        changed_property = 'widget_customization_args'
+                    elif change.content_id == 'default_outcome':
+                        changed_property = 'default_outcome'
+                    elif change.content_id == 'solution':
+                        changed_property = 'solution'
+                    elif change.content_id[:4] == 'hint':
+                        changed_property = 'hints'
+                    elif change.content_id[:8] == 'feedback':
+                        changed_property = 'answer_groups'
+                    elif change.content_id[:10] == 'rule_input':
+                        changed_property = 'answer_groups'
+                    # A condition to store the name of the properties changed
+                    # in changed_properties dict.
+                    state_name = change.state_name
+                    if state_name in new_to_old_state_names:
+                        state_name = new_to_old_state_names[change.state_name]
+                    if state_name in changed_translations:
+                        if (changed_property not in
+                                changed_translations[state_name]):
+                            changed_translations[state_name].append(
+                                changed_property)
+                    else:
+                        changed_translations[state_name] = [changed_property]
+                    if state_name in changed_properties:
+                        if ('written_translations' not in
+                                changed_properties[state_name]):
+                            changed_properties[state_name].append(
+                                'written_translations')
+                    else:
+                        changed_properties[state_name] = [
+                            'written_translations']
+
+            # Old_to_new_state_names: dict. Dictionary mapping state names of
+            # prev_exp_version to the state names of current_exp_version.
+            # It doesn't include the name changes of added/deleted states.
+            old_to_new_state_names = {
+                value: key for key, value in new_to_old_state_names.items()
+            }
+
+            if len(added_state_names) > 0 or len(deleted_state_names) > 0:
+                # Here we will send the changelist, version, latest_version,
+                # and exploration to the admin, so that the conditions
+                # can we reviewed.
```

Comment:
we -> be.
Also, how does the calling function distinguish this specific case, since it returns false later on as well, or are all instances where this function returns false escalated to admins?

Reply:
Done.